app/src/main/java/io/appium/settings/LocationTracker.java

```diff
@@ -138,6 +210,45 @@ synchronized void start(Context context) {
         }
     }
 
+    synchronized void start(Context context, FusedLocationProviderClient fusedLocationProviderClient) {
+        mFusedLocationClient = fusedLocationProviderClient;
+        start(context);
+    }
+
+
+    private FusedLocationProviderClient mFusedLocationClient;
+    private LocationRequest mLocationRequest;
+    private LocationCallback mLocationCallback;
+
+    private void createLocationRequest() {
+        mLocationRequest = new LocationRequest();
+        mLocationRequest.setInterval(UPDATE_INTERVAL_IN_MILLISECONDS);
+        mLocationRequest.setFastestInterval(FASTEST_UPDATE_INTERVAL_IN_MILLISECONDS);
+        mLocationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+    }
+
+    private void initializeRequestLocation(){
+        if (mFusedLocationClient == null){
+            Log.d(TAG, "Failed to initialize request location");
+            return;
+        }
+        createLocationRequest();
+        mLocationCallback = new LocationCallback() {
+            @Override
+            public void onLocationResult(LocationResult locationResult) {
+                super.onLocationResult(locationResult);
+                // This is the result!!!
+                Log.d(TAG, "Frequent request location get update from FusedLocationClient");
+                mLocation = locationResult.getLastLocation();
+            }
+        };
+        try {
+            mFusedLocationClient.requestLocationUpdates(mLocationRequest,
+                    mLocationCallback, Looper.getMainLooper());
+        } catch (SecurityException unlikely) {
+            Log.e(TAG, "Lost location permission. Could not request updates. " + unlikely);
```

Comment:
it makes sense to print full stack trace, so the exception object must be passed as another argument rather than catenated with the message string